CryptoAnalysis/src/main/java/crypto/reporting/ErrorMarkerListener.java

```diff
@@ -54,24 +55,44 @@
 	protected final Table<SootClass, SootMethod, Set<AbstractError>> errorMarkers = HashBasedTable.create(); 
 	protected final Map<Class, Integer> errorMarkerCount = new HashMap<Class, Integer>();
 	protected final List<IAnalysisSeed> secureObjects = new ArrayList<IAnalysisSeed>();
+	protected List<String> ignorePackagesList =  Collections.<String>emptyList();
 	
 	private void addMarker(AbstractError error) {
 		SootMethod method = error.getErrorLocation().getMethod();
 		SootClass sootClass = method.getDeclaringClass();
-		
 		Set<AbstractError> set = errorMarkers.get(sootClass, method);
+		Integer integer = errorMarkerCount.get(error.getClass());
 		if(set == null){
 			set = Sets.newHashSet();
 		}
-		if(set.add(error)){
-			Integer integer = errorMarkerCount.get(error.getClass());
-			if(integer == null){
-				integer = 0;
+		if(integer == null){
+			integer = 0;
+		}
+		if(!ignorePackagesList.isEmpty()) {
+			if(!ignorePackagesList.stream().anyMatch((s -> sootClass.getName().contains(s)))) {
+				if(set.add(error)){	
+					integer++;
+					errorMarkerCount.put(error.getClass(), integer);
+				}
+				errorMarkers.put(sootClass, method, set);
 			}
-			integer++;
-			errorMarkerCount.put(error.getClass(), integer);
 		}
-		errorMarkers.put(sootClass, method, set);
+		else {
+			if(set.add(error)){
+				integer++;
+				errorMarkerCount.put(error.getClass(), integer);
+			}
+			errorMarkers.put(sootClass, method, set);
+		}	
+	}
+	
+/**
+ * This method is used to set the list of packages to be ignored
+ * @param packageList This parameter provides the list of package names to be excluded
+ * @return void.
+*/
+	public void setIgnorePackages(List<String> packageList) {
```

Comment:
This property comes out of no where. The whole class only has message recieve events, while this can mutate the state of adding errors.

I have to see what we can do about this.